pandas/core/reshape/merge.py

```diff
@@ -1200,23 +1200,27 @@ def _maybe_coerce_merge_keys(self) -> None:
 
                 # check whether ints and floats
                 elif is_integer_dtype(rk.dtype) and is_float_dtype(lk.dtype):
-                    if not (lk == lk.astype(rk.dtype))[~np.isnan(lk)].all():
-                        warnings.warn(
-                            "You are merging on int and float "
-                            "columns where the float values "
-                            "are not equal to their int representation.",
-                            UserWarning,
-                        )
+                    # GH 47391 numpy > 1.24 will raise a RuntimeError for nan -> int
```

Comment:
for 1.5 we ought to actually remove the nans first

Reply:
So in 1.5. add a deprecation noting that nans will be dropped?

Reply:
no i mean i think u can remove the nans before comparing to avoid the warning (this is all internal anyhow)

Reply:
Ah gotcha. Yeah can clean this for 1.5 in a separate PR